craft_application/services/lifecycle.py

```diff
@@ -181,6 +189,18 @@ def run(self, step_name: str | None, part_names: list[str] | None = None) -> Non
         target_step = _get_step(step_name) if step_name else None
 
         try:
+            if (
+                self._app.features.package_repository
+                and self._project.package_repositories
+            ):
+                emit.trace("Installing package repositories")
+                package.RepositoryService.install_package_repositories(
```

Comment:
I don't think this `RepositoryService` makes sense as a Service. These are basically two functions that get all parameters they need, why not use two free functions or methos in this `LifecycleService`?

Reply:
Personally I would put it in this `lifecycle` module somewhere (either as staticmethods or free functions) and then if we want to reuse it in the future we can move it
or a `repository` module, but it doesn't need a class

Reply:
OK